### ISSUE TITLE
Always getting complex eigan values. 

### ISSUE THREAD

**Issue by Jeevi10** [liyu10000/DeepLDA#1]

**Comment by liyu10000** [comment]
Have you checked your input features? They need to be non-singular in some sense. A fast way to check is to apply some commercial LDA tools, such as sklearn LDA, on the inputs X and see how it works

**Comment by Jeevi10** [comment]
Thanks, problem was batch size is less than feature size. I fixed it. 
should be Sample_size > feature_size.

**Comment by liyu10000** [comment]
Glad you solved it. One more thing I want to add is if you are working with multiple classes, please make sure each class contains enough samples in every batch so that the decomposition is robust.

**Comment by Jeevi10** [comment]
I trained this with my own model, I used your LDA implementation and trained, I noticed after model converged I always get complex eigan values. Do you have any idea why this occurs ?

**Comment by liyu10000** [comment]
I recommend you pretrain your model with custom loss functions such as bce and then use lda. If you still get into the problem, I guess the problem is the model has not really converged. How is your loss curve? Can you show it once?

**Comment by liyu10000** [comment]
Another quick check on convergence is to pull out of the features of the model, and use some classifiers such svm or lda to do classification. If the classification is bad, it means the model is not trained as expected.
